sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/LocationCache.java

```diff
@@ -314,15 +324,16 @@ private boolean isExcludeRegionsConfigured(List<String> excludedRegionsOnRequest
     }
 
     public URI resolveFaultInjectionEndpoint(String region, boolean writeOnly) {
-        Utils.ValueHolder<URI> endpointValueHolder = new Utils.ValueHolder<>();
+        Utils.ValueHolder<RegionalEndpoints> endpointValueHolder = new Utils.ValueHolder<>();
         if (writeOnly) {
-            Utils.tryGetValue(this.locationInfo.availableWriteEndpointByLocation, region, endpointValueHolder);
+            Utils.tryGetValue(this.locationInfo.availableWriteEndpointsByLocation, region, endpointValueHolder);
         } else {
-            Utils.tryGetValue(this.locationInfo.availableReadEndpointByLocation, region, endpointValueHolder);
+            Utils.tryGetValue(this.locationInfo.availableReadEndpointsByLocation, region, endpointValueHolder);
         }
 
         if (endpointValueHolder.v != null) {
-            return endpointValueHolder.v;
+            // TODO: Figure out how to integrate thinclient into fault injection
```

Comment:
Dito - tracking by fix search-term / owner - so, we cna easily cross check them with ThinClient PR

Reply:
Tracking with - https://github.com/Azure/azure-sdk-for-java/issues/44429